kitsune/questions/views.py

```diff
@@ -1580,3 +1568,23 @@ def _add_tag(request, question_id):
 def _init_watch_form(request, event_type="solution"):
     initial = {"event_type": event_type}
     return WatchQuestionForm(request.user, initial=initial)
+
+
+def _save_and_mark_as_spam(request, object):
+    object.is_spam = True
+    object.save()
+
+    flag = FlaggedObject(
+        content_type=ContentType.objects.get_for_model(object),
+        object_id=object.id,
+        reason='spam',
+        notes="Automatically flagged at creation and marked as spam. \
+            If it's not spam, please reactivate the user's account \
+            and click 'Unmark as spam' on the post to make it publicly visible.",
+        creator=request.user,
+    )
+    flag.save()
+
+    if switch_is_active('blocklist_deactivates_users'):
```

Comment:
Let's remove this completely from here. There is not point in deactivating users if we are to move replies into a moderation queue.